### ISSUE TITLE
feature(engine): read from multiple streams using same event implementation

### ISSUE THREAD

**Issue by leosmerling-hopeit** [hopeit-git/hopeit.engine#35]
In some scenarios where a single stream is used for:
1) Send multiple events to process in a "batch" fashion (i.e. backfilling/reprocess)
2) Run single update events that should be real-time

Both scenarios compete on the stream processing resources, leading to longer wait for real-time events, in the case a backfilling process is running, leading to high processing latency for real-time scenarios:

ORIGINAL PROPOSAL:
- Allow multiple STREAM events to use the same implementation

```
"consume_event.backfill": {
   "type": "STREAM",
   "impl: "consume_event" <-- will use event implemented in consume_event.py
   "read_stream":  {"name": "backfill_stream", "group": "{auto}"}
   "write_stream": "{auto}" <-- will write downstream events to another backfill specific stream
},
"consume_event.realtime": {
   "type": "STREAM",
   "impl: "consume_event" <-- will use event implemented in consume_event.py
   "read_stream":  {"name": "realtime_stream", "group": "{auto}"}
},

```
This will enable backfill/real-time scenarios to use their own streams and be processed concurrently while sharing the same code with no further boilerplate. Engine will spawn independent processes to consume real-time events without being blocked by backfilling/reprocess.

ALTERNATIVES:

-As another option, instead of specifying multiple events pointing to the same implementation, the configuration could support a list of read_stream entries. The disadvantage is that custom stream settings like delay, batch_size and throttling will have to be shared by multiple stream readers, while the approach of configuring separate events allows more fine-grained control.

-Instead of using an `impl` entry in config, event name implementation could be guessed from event name if we use special characters to define event __instances__ (i.e. `$`). In the following example, event file name could be done ignoring what's after $:
```
"consume_event$backfill": {
   "type": "STREAM",
   "read_stream":  {"name": "backfill_stream", "group": "{auto}"}
   "write_stream": "{auto}$backfill" <-- will write downstream events to another backfill specific stream only if $backfill is specified
},
"consume_event$realtime": {
   "type": "STREAM",
   "read_stream":  {"name": "realtime_stream", "group": "{auto}"}
},
```

SELECTED SOLUTION: See #57

**Comment by leosmerling-hopeit** [comment]
This proposal seems not be effective since it could work only when a single stream step is involved, but not when a cascade of events needs to be prioritized.

A way to achieve independent streams for high priority tasks will be to add a "queue" param to each write_stream section. This queue could be used as a suffix to generate the stream_name used to publish events. Then consumers, will need to read and propagate this "queue" values downstream, creating multiple queues for the whole message lifecycle.

Consumer will need to poll from all queues with same stream name (prefix). This would lead to events on the top of each queue to be polled first achieving the effect of "high-priority" consumption on scenario where you have batch processes (in one queue) and real-time events (on other queue), all with same stream name and event processors.

Example config:
```
"consume_event$backfill": {
   "type": "STREAM",
   "read_stream":  {"name": "backfill_stream", "group": "{auto}"}
   "write_stream": "{auto}" <-- will write downstream events to another backfill specific stream only if $backfill is specified
  "queue": "backfill"
},
"consume_event$realtime": {
   "type": "STREAM",
   "read_stream":  {"name": "realtime_stream", "group": "{auto}"}
},
```

**Comment by leosmerling-hopeit** [comment]
An in-progress implementation using 'queues' approach can be found in #57

**Comment by leosmerling-hopeit** [comment]
Closing this issue since the use cases that motivated this enhancement can be solved by #57 

So this is no longer required. It could be reopened if more customization when reading streams is needed.
